### PR TITLE
Add upgrade CI suites for new upgrade tests

### PR DESCRIPTION
**Pull request by krousey**
Creating CI tests for the manual upgrade suites. These tests intentionally don't run a second set of e2es.

### CODE REVIEW

**Comment by krousey** [comment]
(this may have several failures as I bazel doesn't run on ubuntu 16.10)

**Comment by krousey** [comment]
I added "do no merge" label because I will wait for LGTM before creating the GCE projects.

File: jobs/config.json [hunk 1]
```diff
@@ -2493,4 +2493,36 @@
   ]
 }
```
Comment (krzyzacy) [inline jobs/config.json]: },

this is json :-|

File: jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml [hunk 2]
```diff
@@ -1424,6 +1424,33 @@
     # END KUBEMARK
 
     # START UPGRADE
+
+    ### Latest version upgrades #krousey
+    - kubernetes-e2e-gce-latest-upgrade-master: # upgrade master from previous version to latest
+        job-name: ci-kubernetes-e2e-gce-latest-upgrade-master
+        timeout: 70
```
Comment (krzyzacy) [inline jenkins/job-configs/kubernetes-jenkins/bootstrap-ci.yaml]: This timeout is the bootstrap wrapper timeout which needs to be longer than e2e.go timeout which is KUBEKINS_TIMEOUT(90m). Generally we give +20m here to give bootstrap some space to set up repo & upload logs.

File: testgrid/config/config.yaml [hunk 3]
```diff
@@ -869,6 +869,15 @@ test_groups:
 # kubeadm tests
 - name: kubernetes-e2e-kubeadm-gce
   gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-gce
+# upgrade CI tests
+- name: kubernetes-e2e-gce-latest-upgrade-cluster
+  gcs_prefix: kubernetes-jenkins/logs/kubernetes-e2e-gce-latest-upgrade-cluster
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: `gcs_prefix: kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-latest-upgrade-cluster`

**Comment by krzyzacy** [comment]
`self.assertTrue(os.access(job_path, os.X_OK|os.R_OK), job_path)`, emm, maybe it's not required for env files anymore, but for consistency I guess +X to them?

**Comment by krzyzacy** [comment]
And try to update the regex in https://github.com/kubernetes/test-infra/blob/master/jenkins/bootstrap_test.py#L1267-L1275 to make it recognize the json flag.. Sorry we are in the middle of migrating all the upgrade jobs so it's a bit messy here :-p

**Comment by krzyzacy** [comment]
/lgtm

Feel free to merge once you have the gcp projects ready.

**Comment by krousey** [comment]
@krzyzacy I've created the projects and rebased. There were some conflicts that make me think this is no longer correct. Could you take a second pass?

File: jobs/config.json (hunk 1 above)
Comment (krousey) [inline jobs/config.json]: I see how I messed this up.

File: jobs/config.json [hunk 4]
```diff
@@ -2785,195 +2785,35 @@
...
-
-"ci-kubernetes-e2e-gke-gci-1.3-gci-latest-upgrade-cluster": {
-  "scenario": "kubernetes_e2e",
-  "args": [
-    "--env-file=platforms/gke.env",
-    "--env-file=jobs/ci-kubernetes-e2e-gke-gci-1.3-gci-latest-upgrade-cluster.env"
-  ]
-},
-
-"ci-kubernetes-e2e-gke-gci-1.3-gci-latest-upgrade-cluster-new": {
```
Comment (krzyzacy) [inline jobs/config.json]: eh this should not be deleted :thinking:  looks like some rebase issue

File: jenkins/bootstrap_test.py [hunk 5]
```diff
@@ -1281,7 +1281,9 @@ def Check(job, name):
 
     def testBootstrapCIYaml(self):
         # TODO(krzyzacy): temp until more jobs to be converted
-        whitelist = []
+        whitelist = [
+            'kubernetes-e2e-g.e-latest-upgrade-'
```
Comment (krousey) [inline jenkins/bootstrap_test.py]: Is this whitelist entry still appropriate?

Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: yeah that looks fine

File: testgrid/config/config.yaml [hunk 6]
```diff
@@ -1611,6 +1620,17 @@ dashboards:
   - name: gci-1.4-gci-latest-upgrade-master
     test_group_name: kubernetes-e2e-gke-gci-1.4-gci-latest-upgrade-master
 
+- name: latest-upgrades
```
Comment (krzyzacy) [inline testgrid/config/config.yaml]: there's a `google-gke-latest-upgrade` dashboard... what would be the difference? Assume those are 1.5-latest upgrade jobs?

Comment (krousey) [inline testgrid/config/config.yaml]: Yea... there should be some cleanup/consolidation of various "latest" upgrades. I will get around to it... this is just more important right now.

Comment (krzyzacy) [inline testgrid/config/config.yaml]: sgtm. Feel free to merge

File: jenkins/bootstrap_test.py (hunk 5 above)
Comment (krzyzacy) [inline jenkins/bootstrap_test.py]: oh actually I bet you can just remove it since all the upgrade jobs are already converted :-)

**Comment by krousey** [comment]
Trying without the whitelist. Will merge on green.

**Pull request merged**